### DISCUSSION TITLE
Side effects of uninstalling the I2S driver in the AudioOutputI2S.cpp stop() method

### DISCUSSION THREAD

**Discussion by riraosan** [earlephilhower/ESP8266Audio#569] [Q&A]
![2022-09-18_184947](https://user-images.githubusercontent.com/47518585/190951635-2b2d4be2-91d3-49a1-b42d-76539c333231.png)

Hi. everyone. 

about arduino-esp32 2.0.3...

The latest source code uninstalls the I2S driver in the stop() method, as shown in the image above.
This process actually seems to have a side effect.
Uninstalling I2S1=1 is causing the I2S0 driver to be uninstalled.
It looks like a bug in the i2s_driver_uninstall() driver.
How can I work around this? @earlephilhower 

I was using I2S0 for CVBS and I2S1 for Audio, and I noticed that when I stop() Audio, the CVBS (NTSC) signal stops at the same time.

I hope this discussion will be useful for those who use I2S0 and I2S1 at the same time.

**Comment by FedericoBusero** [comment]
As a workaround, you can revert the change https://github.com/earlephilhower/ESP8266Audio/commit/f95f15fcb34e5fad061f0b657b8eac9bdd65a1a2 
However, then you will have 
- issue https://github.com/earlephilhower/ESP8266Audio/issues/561 : error "Unable to install I2S driver"
- issue https://github.com/earlephilhower/ESP8266Audio/issues/488 : possible memory leak in destructor

I suggest another solution:
- I would propose to solve the memory leak in the destructor by introducing another variable next to i2sOn which is true in case the I2S driver has been installed (and doesn't go to false in stop() )
- just ignore the error "Unable to install I2S driver", as it pops up when the driver is already installed
